Make LocaleTranslations KEYS_TO_SKIP configurable

keithpitty opened the pull request:
This PR removes the hard-coded `KEYS_TO_SKIP` array constant from `LocaleTranslations` and makes the array configurable.

keithpitty committed:
Remove hard-coding of keys to skip

Pass keys to skip into constructor of LocaleTranslations rather than
hard-coding the array.

keithpitty committed:
Make keys to skip configurable

* add to Config
* pass config value down to LocaleTranslations via Wrapper
* update specs

keithpitty committed:
Change to keyword argument

keithpitty commented:
![](https://github.trello.services/images/mini-trello-icon.png) [[i18n-hygiene] Make LocaleTranslations KEYS_TO_SKIP configurable](https://trello.com/c/hpxN8BRt/7645-i18n-hygiene-make-localetranslations-keystoskip-configurable)

nickbrowne commented:
Looks like this needs a rebase to resolve some conflicts in `Config` (I'd changed it so you don't have to specify every option unless you really want to).

keithpitty commented:
Rebased.

eleanorakh commented:
I had a brief chat with @keithpitty about this yesterday in slack. When I try to test this in donations I am getting this error: `ArgumentError: missing keyword: keys_to_skip`. I've got the gem file pointing to the latest local copy of `i18n-hygiene`, and I've done a bundle and built up the rake task in donations. Is there something else I need to do? Require a file?

eleanorakh committed:
Pass keys_to_skip to Wrapper
Whilst running the rake task for this PR I was getting undefined method
`keys_to_skip`. As the new behaviour for I18n::Hygiene::Wrapper expects
`keys_to_skip` to be passed into it, I made two changes. The first uses
the configurable behaviour, but the second just passes in an empty array.
I'm not sure if this is the best approach, so I'd welcome a discussion.

yob commented on lib/i18n/hygiene/keys_with_matched_value.rb:
```diff
@@ -6,7 +6,7 @@ class KeysWithMatchedValue
 
       def initialize(regex, i18n_wrapper = nil, reject_keys: nil)
         @regex = regex
-        @i18n = i18n_wrapper || I18n::Hygiene::Wrapper.new
```
It's interesting that the build didn't fail after the `I18n::Hygiene::Wrapper` constructor changed. I'm guessing that's down to the unit specs for these classes using a double rather than a real `I18n::Hygiene::Wrapper`?

nickbrowne replied:
My kingdom for `WrapperTestDouble implements WrapperInterface`

nickbrowne commented:
Rebased on latest master

nickbrowne commented:
:+1:
